pilot/docker/Dockerfile.proxyv2

```diff
@@ -14,12 +14,18 @@ COPY gcp_envoy_bootstrap.json /var/lib/istio/envoy/gcp_envoy_bootstrap_tmpl.json
 RUN chown -R istio-proxy /var/lib/istio
 
 # The following section is used as base image if BASE_DISTRIBUTION=distroless
-FROM gcr.io/distroless/cc@sha256:f81e5db8287d66b012d874a6f7fea8da5b96d9cc509aa5a9b5d095a604d4bca1 as distroless
+FROM gcr.io/distroless/cc-debian10@sha256:5bc7ef365a81ac2a58de8b9606d6840ea1aa4e03847137671671384b0b1700d4 as distroless-amd64
+COPY --from=default /usr/lib/x86_64-linux-gnu/xtables/ /usr/lib/x86_64-linux-gnu/xtables
+COPY --from=default /usr/lib/x86_64-linux-gnu/ /usr/lib/x86_64-linux-gnu
+
+FROM gcr.io/distroless/cc-debian10@sha256:5bc7ef365a81ac2a58de8b9606d6840ea1aa4e03847137671671384b0b1700d4 as distroless-arm64
```

Comment:
instead of having 2x of this, is it possible to just have the copies parameterized on TARGETARCH?

Reply:
TARGETARCH  as [platform arg](https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope) will automatic pass when using `buildx` for multi-arch builds. 

and the values are `amd64` `arm64`
but the arch name in file path is `x86_64` and `aarch64` 

we couldn't pass one single build arg to switch for different args.
so have to 2x build stages, and switch by `TARGETARCH` in `FROM distroless-${TARGETARCH}`.

Reply:
another way is add distroless-prepare stage, and copy deps libs in to `/tmp/lib`.
```Dockerfile 
FROM gcr.io/distroless/cc-debian10@ sha256:5bc7ef365a81ac2a58de8b9606d6840ea1aa4e03847137671671384b0b1700d4 AS distroless-base
FROM default as distroless-prepare

# tmp folder for prepare distroless /usr/lib
WORKDIR /tmp/lib

# copy other files in /usr/lib of cc-debian10
COPY --from=distroless-base /usr/lib/ /tmp/lib/

RUN set -eux; \
    \
    export ARCH=$(uname -m); \
    \
    mkdir -p ./${ARCH}-linux-gnu/xtables/; \
    cp /usr/lib/${ARCH}-linux-gnu/xtables/* ./${ARCH}-linux-gnu/xtables/; \
    cp /usr/lib/${ARCH}-linux-gnu/*\.so* ./${ARCH}-linux-gnu/

FROM distroless-base AS distroless

# TODO(https://github.com/istio/istio/issues/17656) clean up this hack
COPY --from=distroless-prepare /sbin/xtables-multi /sbin/iptables* /sbin/ip6tables* /sbin/ip /sbin/
COPY --from=distroless-prepare /tmp/lib/ /usr/lib/
COPY --from=distroless-prepare /etc/iproute2 /etc/iproute2
```

be more complex, but it will support all arch.

Reply:
@howardjohn Which version do you prefer to?